addons/auth_signup/controllers/main.py

```diff
@@ -39,37 +43,100 @@ def web_auth_signup(self, *args, **kw):
+                            response.headers['Content-Security-Policy'] = "frame-ancestors 'self'"
+                            return response
+                        else:
+                            raise SignupError(_("Another user is already registered using this email address."))
+                    else:
+                        raise UniqueViolation(_("Another temporary is user is already registerd using this email address."))
+                except UniqueViolation:
+                    response = request.render('auth_signup.res_user_signup_pending', qcontext)
+                    response.headers['X-Frame-Options'] = 'SAMEORIGIN'
+                    response.headers['Content-Security-Policy'] = "frame-ancestors 'self'"
+                    return response
+                except UserError as e:
+                    qcontext['error'] = e.args[0]
+                except (SignupError, AssertionError) as e:
+                    if request.env["res.users"].sudo().search([("login", "=", qcontext.get("login"))]):
+                        qcontext["error"] = _("Another user is already registered using this email address.")
+                    else:
+                        _logger.error("%s", e)
+                        qcontext['error'] = _("Could not create a new account.")
+
+            elif 'error' not in qcontext and 'login' in qcontext:
+                signup_user = request.env['res.users.signup'].sudo().search([('login', '=', qcontext['login'])])
+                if signup_user:
+                    success = signup_user.send_mail()
+                    if success:
+                        qcontext['success'] = _("A new mail has been sent")
+                    else:
+                        qcontext['error'] = _("Wait a little before sending more mails.")
+
+                    response = request.render('auth_signup.res_user_signup_pending', qcontext)
+                    response.headers['X-Frame-Options'] = 'SAMEORIGIN'
+                    response.headers['Content-Security-Policy'] = "frame-ancestors 'self'"
+                    return response
+
+            elif 'error' not in qcontext and 'signup_email' in qcontext:
+                user = request.env['res.users'].sudo().search([('email', '=', qcontext.get('signup_email')), ('state', '!=', 'new')], limit=1)
+                if user:
+                    return request.redirect('/web/login?%s' % url_encode({'login': user.login, 'redirect': '/web'}))
 
         response = request.render('auth_signup.signup', qcontext)
         response.headers['X-Frame-Options'] = 'SAMEORIGIN'
         response.headers['Content-Security-Policy'] = "frame-ancestors 'self'"
         return response
 
+    @http.route('/web/confirm', type='http', auth='public', website=True, sitemap=False)
+    def web_auth_confirm(self, *args, **kw):
+        if kw.get('token'):
+            signup_user = request.env['res.users.signup'].sudo().search([('token', '=', kw['token'])])
```

Comment:
wondering if we should also pass the login and check the token matches the one on the user :thinking: